addons/point_of_sale/models/pos_config.py

```diff
@@ -360,6 +345,8 @@ def create(self, values):
         pos_config.sudo()._check_modules_to_install()
         pos_config.sudo()._check_groups_implied()
         # If you plan to add something after this, use a new environment. The one above is no longer valid after the modules install.
+        if not values.get('barcode_nomenclature_id'):
+            values["barcode_nomenclature_id"] = self.env.user.company_id.nomenclature_id
```

Comment:
why not use a default value on the field? Or a related if we want to force one (but I see this more as a default)

Reply:
I like to add unnecessary complexity. But default value sounds ok too ;)